src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -475,6 +477,29 @@ public function resolveConcatType(Type $left, Type $right): Type
 			$accessoryTypes[] = new AccessoryLiteralStringType();
 		}
 
+		$leftNumericStringNonEmpty = TypeCombinator::remove($leftStringType, new ConstantStringType(''));
+		if ($leftNumericStringNonEmpty->isNumericString()->yes()) {
+			$allRightConstantsZeroOrMore = false;
+			foreach ($rightConstantStrings as $rightConstantString) {
+				if ($rightConstantString->getValue() === '') {
+					continue;
+				}
+
+				if (!is_numeric($rightConstantString->getValue()) || $rightConstantString->getValue() < 0) {
+					$allRightConstantsZeroOrMore = false;
+					break;
+				}
+
+				$allRightConstantsZeroOrMore = true;
```

Comment:
This is going to pass for floats too. `$numericString . 0.2` will not be `numeric-string`. https://3v4l.org/BCS2K

Reply:
good catch, added more tests and a fix